lib/waistband/model.rb

```diff
@@ -0,0 +1,195 @@
+require 'digest/md5'
+require 'active_support/core_ext/object/blank'
+require 'active_model'
+
+module Waistband
+  class Model
+
+    extend ::ActiveModel::Callbacks
```

Comment:
I think you might be able to just use ActiveModel::Model but it's been a while since I've used it.